book_config.py

```diff
@@ -130,75 +111,43 @@ def __init__(self, db, ids, parent):
     @property
     def book(self):
         return self._book_settings[self._index]
-    
-    def edit_asin(self, val):
-        self.book.asin = val
 
-    def edit_shelfari_url(self, val):
-        http_string = 'http://www.'
+    def edit_goodreads_url(self, val):
```

Comment:
What is the aim of this?

I'm not sure there is much benefit in policing URLs to fit a certain format - what if someone enters it without the www. subdomain?  What if there is a beta.goodreads.com?  And so on.  If the URL fetches and parses ok, that should be good enough IMO.

If you think it is worthwhile then there is something that needs fixing here - as it will just do weird things in certain cases (any time you manually edit anything in "https://www.goodrads.com" rather than pasting in a URL).  For instance: if I remove the sub-doman "www" from the URL, I'll end up with "https://www.goodreads.com/goodreads.com/foobar" which is obviously garbage.